### ISSUE TITLE
Rename deconvolution and identification to model_estimation

### ISSUE THREAD

**Issue by BjoernLudwigPTB** [PTB-M4D/PyDynamic#31]
To avoid unambiguous method naming we will **combine** all methods from ***identification*** and ***devonvolution*** **in one** new module ***model_estimation*** including the *fit_filter.py* methods with same namings in both modules. That requires us to rename some methods out of *devonvolution.fit_filter.py*. Because of the following incompatibility with previous versions of PyDynamic we need to insert a deprecation warning into the next minor release and inform users about the upcoming change.

**Comment by BjoernLudwigPTB** [comment]
Start with [deprecation warning](http://www.jaggedverge.com/2016/09/deprecation-warnings-in-python/).

**Comment by BjoernLudwigPTB** [comment]
@eichstaedtPTB What about [identification/fit_transfer.py](https://github.com/eichstaedtPTB/PyDynamic/blob/master/PyDynamic/identification/fit_transfer.py), should that be moved to the new module *model_estimation* as well or should we think about something else?

**Comment by eichstaedtPTB** [comment]
> 
> 
> @eichstaedtPTB What about [identification/fit_transfer.py](https://github.com/eichstaedtPTB/PyDynamic/blob/master/PyDynamic/identification/fit_transfer.py), should that be moved to the new module _model_estimation_ as well or should we think about something else?

This should be moved into the new module, too.

**Comment by eichstaedtPTB** [comment]
The methods that fit the reciprocal could be named "iLSFIR", "iLSIIR", etc.

**Comment by BjoernLudwigPTB** [comment]
@eichstaedtPTB What do you think about the new structure? Should we maybe have a short conversation about it or can you tell from looking [at the code](https://github.com/PTB-PSt1/PyDynamic/blob/introduce_model_estimation/PyDynamic/model_estimation/fit_filter.py), that we can use this new structure for replacing _identification_ and _deconvolution_?

What I did now is prefix the "old" deconvolution methods with "inv" and combine both _fit_filter_ modules in _model_estimation.fit_filter_ and of course transfer _identification.fit_transfer_ into _model_estimation.fit_transfer_

**Comment by BjoernLudwigPTB** [comment]
Introducing these changes we noticed, that we got those `invLS[I,F]IR` with and without uncertainties and Monte Carlo methods. We should merge these implementations and introduce optional parameters to choose one or the other.

This applies to

- invLS[F,I]IR
- invLS[F,I]IR_unc
- invLSFIR_uncMC

There is no `invLSIIR_uncMC` since the uncertainties are determined using MC already in `invLSIIR_unc` .

Besides that we actually do not fit a second-order system but rather a second-order model, thus we should rename `fit_transfer.fit_sos()` to `fit_transfer.fit_som()`.

We should then for half a year after releasing 2.0.0 backport bug fixes to the 1.4 minor releases before we stop supporting these. The docs for 2.0.0 should be enriched with the information on how to tell _pip_ to upgrade only to the newest version of the 1.4 minor release series.
